### ISSUE TITLE
Dependency Target Name

### ISSUE THREAD

**Issue by timothymcgrath** [Microsoft/ApplicationInsights-SDK-Labs#147]
I'd like to set the dependency target name to the full wcf uri instead of just the host name. All of my wcf services live behind the same host name, so they all get grouped together as the same dependency. Since each service has a unique uri, it'd be helpful to set the target to the full uri instead.

Here's the code in ClientTelemetryChannelBase.cs where it sets the target to Uri.Host: 
```
                var telemetry = new DependencyTelemetry();
                this.ChannelManager.TelemetryClient.Initialize(telemetry);

                telemetry.Start();
                telemetry.Type = DependencyConstants.WcfClientCall;
                telemetry.Target = this.RemoteAddress.Uri.Host;
```
Is there any reason to not eliminate the .Host? Since the wcf uri doesn't change per operation, it seems like it'd make more sense.

Thanks! I really like this library so far!

**Comment by timothymcgrath** [comment]
@tomasr Right, the built-in one does it that way, too, which makes sense since it wouldn't know which part of the uri was the service and which part was just an operation on the service. It would end up with a dependency for every operation.

But since WCF usually has a single uri, it seems like it makes sense. But maybe I just have a special case where I have a lot of small WCF services and all my calls go to a single domain which gets rerouted by HAProxy.

I hooked in a new TelemetryProcessor based on the idea I saw to fix the double HTTP dependency tracking in Issue #105. That worked great to rename the Target to the correct URL based on what is set in Data. It looks better in the Azure Search screen, and it groups them correctly in the new Preview Application Map now. I have 6 clear WCF dependencies on the map instead of just 1 named "http://backendservices-qa/".
